src/hints/execution.rs

```diff
@@ -1470,25 +1441,23 @@ pub fn write_syscall_result(
     Ok(())
 }
 
-pub const GEN_NONCE_ARG: &str = indoc! {r#"
-	ids.tx_version = tx.version
-	ids.max_fee = tx.max_fee
-	ids.sender_address = tx.sender_address
-	ids.calldata = segments.gen_arg([tx.class_hash])
-
-	if tx.version <= 1:
-	    assert tx.compiled_class_hash is None, (
-	        "Deprecated declare must not have compiled_class_hash."
-	    )
-	    ids.additional_data = segments.gen_arg([tx.nonce])
-	else:
-	    assert tx.compiled_class_hash is not None, (
-	        "Declare must have a concrete compiled_class_hash."
-	    )
-	    ids.additional_data = segments.gen_arg([tx.nonce, tx.compiled_class_hash])"#
+pub const GEN_CLASS_HASH_ARG: &str = indoc! {r#"
+    ids.tx_version = tx.version
+    ids.sender_address = tx.sender_address
+    ids.class_hash_ptr = segments.gen_arg([tx.class_hash])
+    if tx.version <= 1:
+        assert tx.compiled_class_hash is None, (
+            "Deprecated declare must not have compiled_class_hash."
+        )
+        ids.compiled_class_hash = 0
+    else:
+        assert tx.compiled_class_hash is not None, (
+            "Declare must have a concrete compiled_class_hash."
+        )
+        ids.compiled_class_hash = tx.compiled_class_hash"#
```

Comment:
This type of renaming should be in its own PR.

Reply:
Not sure what you mean, the functionality of the hint changed -- it now has nothing to do with any nonce...